src/Logary/Configuration_Config.fs

```diff
@@ -140,40 +137,48 @@ let validate ({ targets  = targets
 [<CompiledName "RunLogary"; Extension>]
 let runLogary conf =
   let instance = Advanced.create conf
-  Logging.startFlyweights instance
-  instance
+  Logging.startFlyweights instance >>-. instance
 
 /// Shutdown logary, waiting maximum flushDur + shutdownDur.
 [<CompiledName "ShutdownLogary">]
-let shutdown' (flushDur : Duration) (shutdownDur : Duration)
-  ({ registry = reg; metadata = { logger = lgr } } : LogaryInstance)
-  = 
-  let log = LogLine.setPath "Logary.Configuration.Config.shutdown" >> Logger.log lgr
-  async {
-  LogLine.info "start shutdown" |> log
-  let! res = Advanced.flushAndShutdown flushDur shutdownDur reg
-  LogLine.info "stop shutdown" |> log
-  Logging.shutdownFlyweights ()
-  shutdownLogger lgr
-  return res
+let shutdown (flushDur : Duration) (shutdownDur : Duration) (inst : LogaryInstance) =
+  let log =
+    Message.setName (PointName ["Logary"; "Configuration"; "Config"; "shutdown"])
+    >> Logger.log inst.runtimeInfo.logger
+
+  job {
+    do! Message.info "start shutdown" |> log
+    let! res = Advanced.flushAndShutdown flushDur shutdownDur inst.registry
+    do! Message.info "stop shutdown" |> log
+    Logging.shutdownFlyweights ()
+    shutdownLogger inst.runtimeInfo.logger
+    return res
   }
 
 /// Shutdown logary, waiting maximum 30 seconds, 15s for flush and 15s for
 /// shutdown.
 [<CompiledName "ShutdownLogary"; Extension>]
-let shutdown =
-  shutdown' (Duration.FromSeconds 15L) (Duration.FromSeconds 15L)
+let shutdownSimple =
+  shutdown (Duration.FromSeconds 15L) (Duration.FromSeconds 15L)
 
 /// Wrap the LogaryInstance as a LogManager
 [<CompiledName "AsLogManager"; Extension>]
 let asLogManager (inst : LogaryInstance) =
-  let run = Async.RunSynchronously
   { new LogManager with
-      member x.RuntimeInfo        = inst.metadata
-      member x.GetLogger name     = name |> getLogger inst.registry |> run
-      member x.FlushPending dur   = Advanced.flushPending dur inst.registry |> run
-      member x.Shutdown fDur sDur = shutdown' fDur sDur inst |> run
-      member x.Dispose ()         = shutdown inst |> Async.Ignore |> run
+      member x.runtimeInfo =
+        inst.runtimeInfo
+
+      member x.getLogger name =
+        name |> getLogger inst.registry |> run
+
+      member x.flushPending dur =
+        Advanced.flushPending inst.registry <|> timeOut (dur.ToTimeSpan ())
```

Comment:
Note that is usually preferable to have `Alt` operations without baked in timeouts.  In other words, instead of `flushPending: Duration -> Alt<unit>` you could have `flushPending: Alt<unit>`.  The user of `flushPending` would still be able to combine with a `timeOut` or run the operation without a timeout.